doc/user/flowspec.rst

```diff
@@ -0,0 +1,345 @@
+
+- A configuration of each VRF must be done, with its Route Target set
+
+Each VRF is being configured within a BGP VRF instance with its own Route Target list.
+Route Target accepted format matches the following: ``A.B.C.D:U16``, or ``U16:U32``,
+``U32:U16``.
+
+- The first VRF with the matching Route Target will be selected to route traffic to. Use
+  the following command under ipv4 unicast address-family node
+
+.. index:: [no] rt redirect import RTLIST...
+.. clicmd:: [no] rt redirect import RTLIST...
+
+In order to illustrate, if the Route Target configured in the Flowspec entry is E.F.G.H:II,
+then a BGP VRF instance with the same Route Target will be set set. That VRF will then be
+selected. The below full configuration example depicts how Route Targets are configured
+and how VRFs and cross VRF configuration is done. Note that the VRF are mapped on Linux
+Network Namespaces. For data traffic to cross VRF boundaries, virtual ethernet interfaces
+are created with private IP adressing scheme.
+
+.. code-block:: frr
+
+   router bgp <ASx>
+    neighbor <A.B.C.D> remote-as <ASz>
+    address-family ipv4 flowspec
+     neighbor A.B.C.D activate
+    exit
+   exit
+   router bgp <ASy> vrf vrf2
+    address-family ipv4 unicast
+     rt redirect import <E.F.G.H:II>
+    exit
+   exit
+
+Flowspec Monitor and troubleshooting
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can monitor policy-routing objects by using one of the following commands.
+Those command rely on the filtering contexts configured from BGP, and get the
+statistics information retrieved from the underlying system. In other words, those
+statistics are retrieved from ``Netfilter``.
+
+.. index:: show pbr ipset IPSETNAME | iptable
+.. clicmd:: show pbr ipset IPSETNAME | iptable
+
+``IPSETNAME`` is the policy routing object name created by ``ipset``. 
+About rule contexts, it is possible to know which rule has been configured to
+policy-route some specific traffic. The :clicmd:`show pbr iptable` command displays for
+forwarded traffic, which table is used. Then it is easy to use that table identifier
+to dump the routing table that the forwarded traffic will match.
+
+.. code-block:: frr
+
+   show ip route table TABLEID
+
+``TABLEID`` is the table number identifier referencing the non standard routing table
+used in this example.
+You can troubleshoot Flowspec, or BGP policy based routing. For instance, if you
+encounter some issues when decoding a Flowspec entry, you should enable :clicmd:`debug bgp
+flowspec`.
+
+.. index:: [no] debug bgp flowspec
+.. clicmd:: [no] debug bgp flowspec
+
+If you fail to apply the flowspec entry into *zebra*, there should be some
+relationship with policy routing mechanism. Here, :clicmd:`debug bgp pbr error`
+could help.
+
+.. index:: [no] debug bgp pbr error
+.. clicmd:: [no] debug bgp pbr error
+
+To get information about policy routing contexts created/removed, use the following
+command:
```

Comment:
Just write a reference to the command here instead of defining it again.

Reply:
I use :clicmd:`debug bgp pbr` reference. is it ok?